### ISSUE TITLE
element_checkbox.erl

### ISSUE THREAD

**Issue by sgibberd** [5HT/n2o#69]
Hi,

I tried the following
# checkbox{id=admin, checked=Admin, body= <<"Administrator">>},
# button{ body = <<"Save">>, postback = save,source = [name,password,admin]}

but wf:q(admin) always equals 'on' - the default value whether it is checked or not. My first attempt to correct this was:
# panel{id="check_" ++Id, body=[#checkbox{id=admin, checked=Admin, value=Admin, body= <<"Administrator">>, postback={setvalue,admin,not Admin}}]}

and then to handle the postback

``` erlang
event({setvalue,Id,Val}) ->
    wf:update("check_" ++ Id, #panel{id= "check_" ++ Id, body=[#checkbox{id=Id,\
 checked=Val, body=Id, postback={setvalue,Id,not Val}, value=Val}]});
```

while it worked, it seems overly complex. Then I modified action_event.erl to

``` Erlang
render_action(#event{postback=Postback,actions=Actions,source=Source,target=Control,
                                   type=Type,delegate=Delegate}) ->
    Data = "[" ++ string:join([begin
        {Key, Id} = if  is_atom(Src)-> S = atom_to_list(Src),
                        {"atom('"++S++"')", S};
                    true -> {"utf8.toByteArray('" ++ Src ++ "')", Src} end,
        "tuple(" ++ Key ++ ", document.querySelector('#" ++ Id ++ "').type == 'checkbox' ?
          utf8.toByteArray(document.querySelector('#" ++ Id ++ "').checked.toString()) :  
           utf8.toByteArray(document.querySelector('#"  ++ Id ++ "').value))" end || Src <- Source]
       ++ ["tuple(tuple(utf8.toByteArray('"++ Control ++"'), bin('detail')), event.detail)"],",") ++ "]",
    PostbackBin = wf_event:new(Postback, Control, Delegate, event, Data),
    [wf:f("document.querySelector('#~s').addEventListener('~s',function (event) {", [Control,Type]),PostbackBin,"});"].
```

Basically testing if the element is a checkbox, and returning the value of 'checked' instead of 'value'. This implements the behaviour I was expecting for the checkbox. Am I missing something on the checkboxes?

Thanks, Stephen
